### PR TITLE
Added tests covering TSDB read path in the ingester

### PR DESCRIPTION
**Pull request by pracucci**
**What this PR does**:
While working on PR #1830 I've realized we don't have any test on the TSDB read path in the ingester. In this PR I'm adding them.

**Which issue(s) this PR fixes**:
_No issue for this_

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by pracucci** [comment]
- @thorfour and @codesome may you take a look, please?
- @tomwilkie may you add this PR to https://github.com/cortexproject/cortex/projects/4 , please?

File: pkg/ingester/ingester_v2_test.go [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+		lbls      labels.Labels
+		value     float64
+		timestamp int64
+	}{
+		{labels.Labels{{Name: labels.MetricName, Value: "test_1"}, {Name: "status", Value: "200"}, {Name: "route", Value: "get_user"}}, 1, 100000},
+		{labels.Labels{{Name: labels.MetricName, Value: "test_1"}, {Name: "status", Value: "500"}, {Name: "route", Value: "get_user"}}, 1, 110000},
+		{labels.Labels{{Name: labels.MetricName, Value: "test_2"}}, 2, 200000},
+	}
+
+	expected := []string{"__name__", "status", "route"}
```
Comment (codesome) [inline pkg/ingester/ingester_v2_test.go]: The result of `LabelNames()` is always sorted in Prometheus. If we were to follow that in Cortex too, then this should be fixed.

File: pkg/ingester/ingester_v2_test.go [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+	// Push series
+	ctx := user.InjectOrgID(context.Background(), "test")
+
+	for _, series := range series {
+		req, _ := mockWriteRequest(series.lbls, series.value, series.timestamp)
+		_, err := i.v2Push(ctx, req)
+		require.NoError(t, err)
+	}
+
+	// Get label values
```
Comment (codesome) [inline pkg/ingester/ingester_v2_test.go]: Wrong comment, I guess that was a copy paste from above :)

File: pkg/ingester/ingester_v2_test.go [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+	expectedResponse := &client.QueryResponse{
+		Timeseries: []client.TimeSeries{
+			{
+				Labels:  client.FromLabelsToLabelAdapters(lbls),
+				Samples: samples,
+			},
+		},
+	}
+
+	return req, expectedResponse
```
Comment (codesome) [inline pkg/ingester/ingester_v2_test.go]: I don't see `expectedResponse` being used anywhere

File: pkg/ingester/ingester_v2_test.go (hunk 2 above)
Comment (pracucci) [reply pkg/ingester/ingester_v2_test.go]: Ops! Yes, I've copied previous test as template. Fixed.

File: pkg/ingester/ingester_v2_test.go (hunk 3 above)
Comment (pracucci) [reply pkg/ingester/ingester_v2_test.go]: This function is a copy and paste from https://github.com/cortexproject/cortex/pull/1830 (still under review). Keeping the copy and paste will help me rebasing once one of the two PRs will get merged.

File: pkg/ingester/ingester_v2_test.go (hunk 1 above)
Comment (pracucci) [reply pkg/ingester/ingester_v2_test.go]: Yes, but this is the ingester, not the querier, so ordering is not required at this stage. The querier fetch label names from all ingesters, and then sort it. Technically it's implemented in the distributor (which runs as internal dependency of the querier) and the [specific code is here](https://github.com/cortexproject/cortex/blob/master/pkg/distributor/distributor.go#L499).

_I'm going to resolve this conversation, but feel free to re-open it if something is missing._

Comment (codesome) [reply pkg/ingester/ingester_v2_test.go]: Ah got it!

**Comment by pracucci** [comment]
@gouthamve I've fixed the conflicts after rebasing. May you help me moving forward with this PR, please?

**Pull request merged**